packages/apollo-federation/src/composition/utils.ts

```diff
@@ -447,3 +448,43 @@ export const defKindToExtKind: { [kind: string]: string } = {
   [Kind.ENUM_TYPE_DEFINITION]: Kind.ENUM_TYPE_EXTENSION,
   [Kind.INPUT_OBJECT_TYPE_DEFINITION]: Kind.INPUT_OBJECT_TYPE_EXTENSION,
 };
+
+export function flat<T>(list: Array<T>[]): T[] {
+  return list.reduce(
+    (array, entry) => {
+      if (Array.isArray(entry)) {
+        array.push(...entry);
+      } else {
+        array.push(entry);
+      }
+      return array;
+    },
+    [] as T[],
+  );
+}
+
+export function flatMap<T, U>(
+  array: T[],
+  callbackfn: (value: T, index: number, array: T[]) => ArrayLike<U>,
+): U[] {
+  return Array.prototype.concat(...array.map(callbackfn));
+}
+
+export function mapValues<T, U = T>(
```

Comment:
@trevor-scheer Not suggesting that they're wrong, but I'm just curious where these `mapValues`, `flatMap` and `flat` poly-fill implementations came from.

Reply:
This is a great point, and I appreciate you bringing it up. @abernix and I spoke offline extensively about this. Our goal was to continue using `core-js`'s implementation without the polyfill. This proved to be very challenging, as `core-js-pure` (no polyfill utility fns) has no associated types. Typing them (even using the stdlib typings) proved to be quite difficult.

The result of this exploration was to:
* Continue using the `core-js` polyfills (but now directly, instead of through `apollo-env`
* Use the `es2019.array` lib for our TS config within the `@apollo/federation` project

Reply:
Semi-related: `mapValues` was already a utility function which I've just lifted from `apollo-env` directly.